docs/contribute/development/ide/vscode.rst

```diff
@@ -0,0 +1,112 @@
+..
+   # *******************************************************************************
+   # Copyright (c) 2025 Contributors to the Eclipse Foundation
+   #
+   # See the NOTICE file(s) distributed with this work for additional
+   # information regarding copyright ownership.
+   #
+   # This program and the accompanying materials are made available under the
+   # terms of the Apache License Version 2.0 which is available at
+   # https://www.apache.org/licenses/LICENSE-2.0
+   #
+   # SPDX-License-Identifier: Apache-2.0
+   # *******************************************************************************
+
+VS Code
+#######
+
+For VS Code any preexisting configuration is provided via the workspace configuration in the directory <workspace_root/.vscode>. It includes
+
+* some formatting rules
+* configuration for recommended plugins
+
+Starlark
+========
+
+For Starlark currently no solution is integrated in VScode.
+
+C++
+===
+
+
+Python
+======
+
+For Python currently the full feature set of the official `Python <https://marketplace.visualstudio.com/items?itemName=ms-python.python>`_  Extension (MIT License) is provided included features e.g. autocomplete, syntax checking, ...
+
+Also the configuration for displaying and executing pytest in the IDE is provided.
+
+As a linter the `ruff extension <https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff>`_  (MIT License) is set up.
+
+Debugging
+---------
+
+For debugging the `Python Debugger <https://marketplace.visualstudio.com/items?itemName=ms-python.debugpy>`_ (MIT License) and *debugpy* can be used for remote debugging.
+
+Debugging the docs build
+''''''''''''''''''''''''
+For debugging the docs environment all necessary configuration is already provided. It can be activated for the incremental build via the following command:
+
+.. code-block:: shell
+
+   bazel run //docs:incremental -- --debug
+
+Manual debugging
+''''''''''''''''
+For all other tools the debugging needs to be setup manually. Therefore following steps need to be executed:
```

Comment:
```suggestion
For all other tools the debugging should work out of the box via VS Code Python extension.

In rare cases where this does not fit, mostly because you want to start the tool via bazel, you can reuse the debugpy approach from docs:incremental:
```